### ISSUE TITLE
submitting data to the server

### ISSUE THREAD

**Issue by joelmccracken** [noahzgordon/elm-jsonapi-http#2]
Hi,

I've been working with your library for a little bit, and just realized that it doesn't support http POST/PUT/PATCH, etc. 

Any thoughts about adding them?

**Comment by noahzgordon** [comment]
Hey Joel! I unfortunately haven't touched this library for a little while since I put in some initial work, but I'm looking to pick it back up soon. Creating/Updating resources is definitely high up on the feature list.

**Comment by noahzgordon** [comment]
(Also sort of waiting for Elm 0.18 so I can anticipate the HTTP library changes)

**Comment by joelmccracken** [comment]
yep, i also noticed that there aren't any corresponding components for encoding things to jsonapi format. 

I need this for what i'm doing, so I may take a stab at them both.

**Comment by noahzgordon** [comment]
@joelmccracken by encoding things, do you mean taking your Elm data type and wrapping it in a `Resource` with links, relations, etc? Is your use case here to POST a brand new resource to the server?

I have to think through the ideal API for constructing / updating resources and POST/PUTing them over HTTP. If you'd like to take a stab and offer suggestions that's great, but my suspicion is that our approaches will differ and that the changes to the HTTP API in 0.18 will change things quite a bit.

**Comment by joelmccracken** [comment]
My use case is to post new resources, and update existing resources. I need
to do both.

I'm fine with that -- i don't need them to be the same.

Also, elm .18 has been released!
